src/candid_rpc.rs

```diff
@@ -221,16 +220,16 @@ impl CandidRpcClient {
 
     pub async fn eth_get_transaction_receipt(
         &self,
-        hash: String,
-    ) -> MultiRpcResult<Option<candid_types::TransactionReceipt>> {
-        match Hash::from_str(&hash) {
-            Ok(hash) => process_result(
-                RpcMethod::EthGetTransactionReceipt,
-                self.client.eth_get_transaction_receipt(hash).await,
-            )
-            .map(|option| option.map(|r| r.into())),
-            Err(_) => MultiRpcResult::Consistent(Err(ValidationError::InvalidHex(hash).into())),
-        }
+        hash: Hex32,
```

Comment:
Strictly speaking this is a change of behaviour:
1. before: if the caller would pass an invalid transaction hash, the error `ValidationError::InvalidHex(hash)` would be returned
2. Now the call would not even make it to `eth_get_transaction_receipt`, because deserialization of the arguments would fail (caller would receive another error)

I have the impression that this is ok, while simplifying the logic a bit. WDYT?

Reply:
This change seems good to me.